parquet-variant-compute/src/type_conversion.rs

```diff
@@ -60,6 +65,44 @@ impl_primitive_from_variant!(datatypes::UInt64Type, as_u64);
 impl_primitive_from_variant!(datatypes::Float16Type, as_f16);
 impl_primitive_from_variant!(datatypes::Float32Type, as_f32);
 impl_primitive_from_variant!(datatypes::Float64Type, as_f64);
+impl_primitive_from_variant!(
+    datatypes::Date32Type,
+    as_naive_date,
+    Date32Type::from_naive_date
+);
+
+pub(crate) trait TimestampFromVariant: ArrowTimestampType {
+    fn from_variant(variant: &Variant<'_, '_>) -> Option<Self::Native>;
+}
+
+macro_rules! impl_timestamp_from_variant {
+    ($timestamp_type:ty, {
+        $(($variant_pattern:pat, $conversion:expr)),+ $(,)?
+    }) => {
```

Comment:
I'm pretty sure this works?

```suggestion
    (
        $timestamp_type:ty,
        $( $variant_pattern:pat => $conversion:expr ),+ $(,)?
    ) => {
```

Use sites would look like this:
```rust
impl_timestamp_from_variant!(
    TimestampMicrosecondType,
    Variant::TimestampMicros(t) => Some(t.timestamp_micros()),
    Variant::TimestampNtzMicros(t) => Some(t.and_utc().timestamp_micros()),
);
```